packages/protocol/scripts/foundry/verify-bytecode.ts

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env ts-node
+/* eslint-disable no-console */
+
+import { verifyBytecodesFoundry } from '@celo/protocol/lib/compatibility/verify-bytecode-foundry'
+import { getFoundryBuildArtifacts } from '@celo/protocol/lib/foundry-artifacts'
+import { CeloContractName, celoRegistryAddress } from '@celo/protocol/lib/registry-utils'
+import { readJsonSync, writeJsonSync } from 'fs-extra'
+import Web3 from 'web3'
+import yargs from 'yargs'
+import { hideBin } from 'yargs/helpers'
+import { getReleaseVersion } from '../../lib/compatibility/ignored-contracts-v9'
+
+/*
+ * Foundry-based bytecode verification script.
+ *
+ * This script verifies that a given set of smart contract bytecodes corresponds
+ * to a Celo system deployed to a given network. It uses the Registry contract
+ * as its source of truth, potentially modified by an optional contract upgrade
+ * proposal description.
+ *
+ * Key differences from Truffle version:
+ * - Uses Foundry artifacts from out-truffle-compat/ directories
+ * - Handles Foundry's hashed library placeholders
+ * - Compiles using forge with truffle-compat profiles
+ * - Runs standalone with ts-node (no Truffle dependency)
+ *
+ * Expects the following flags:
+ *   --build_artifacts: The directory in which smart contract build artifacts
+ *   can be found (default: "./out-truffle-compat/")
+ *   --proposal: The JSON file containing a Governance proposal that
+ *   repoints the Registry to newly deployed Proxies and/or repoints existing
+ *   Proxies to new implementation addresses.
+ *   --initialize_data: The JSON file containing, for each newly deployed Proxy,
+ *   the calldata to its logic contract's `initialize` function.
+ *   --network: The network to verify (e.g., "mainnet", "alfajores", "baklava")
+ *   --rpc_url: The RPC URL to connect to (required)
+ *   --librariesFile: The file to which linked library addresses will be
+ *   written (default: "libraries-foundry.json").
+ *   --branch: The branch name for determining release version.
+ *
+ * Run using ts-node:
+ * yarn ts-node scripts/foundry/verify-bytecode.ts \
+ *   --network alfajores \
+ *   --rpc_url https://alfajores-forno.celo-testnet.org \
+ *   --build_artifacts out-truffle-compat \
+ *   --proposal proposal.json
+ */
+
+interface VerifyBytecodeArgs {
+  build_artifacts?: string
+  proposal?: string
+  initialize_data?: string
+  network?: string
+  rpc_url: string
+  librariesFile?: string
```

Comment:
It looks strange to mix underscores (build_artifacts, rpc_url) & camelCase (librariesFile)
```suggestion
  build_artifacts?: string
  proposal?: string
  initialize_data?: string
  network?: string
  rpc_url: string
  libraries_file?: string
```